src/systems/filecoin_nodes/clock/_index.md

```diff
@@ -3,31 +3,41 @@ title: Clock
 statusIcon: ⚠️
 ---
 
-```go
-type Time string // ISO nano timestamp
-type UnixTime int // unix timestamp
-type Round int // Blockchain round
-
-type Clock interface {
-  UTCTime() Time
-  UnixNano() UnixTime
-
-  CurrentRound() Round
-  LastRoundObserved() Round
-}
-```
-
-TODO:
-
-- explain why we need a system clock
-- explain where it is used
-  - for rejecting/accepting blocks
-- explain synchrony requirements
-  - small clock drift -- <2s
-  - very important to have accurate time
-- explain how we can resync
-  - Network
-    - recommend various NTP servers
-  - Cesium clocks
-- Future work:
-  - VDF Clocks
+{{< readfile file="clock_subsystem.id" code="true" lang="go" >}}
+{{< readfile file="clock_subsystem.go" code="true" lang="go" >}}
+
+Filecoin assumes weak clock synchrony amongst participants in the system. That is, the system relies on participants having access to a globally synchronized clock, tolerating bounded delay in honest clock lower than epoch time (more on this in a forthcoming paper).
+
+Filecoin relies on this system clock in order to secure consensus, specifically ensuring that participants are only running leader elections once per epoch and enabling miners to catch such deviations from the protocol. Given a system start and epoch time by the genesis block, the system clock allows miners to associate epoch and wall clock time, thereby enabling them to reason about block validity and give the protocol liveness.
+
+## Clock uses
+Specifically, the Filecoin system clock is used:
+- to validate incoming blocks and ensure they were mined in the appropriate round, looking at the wall clock time in conjunction with the block's `ElectionProof` (which contains the epoch number) (see {{<sref leader_election>} and {{<sref block_validation>}}).
+- to help protocol convergence by giving miners a specific cutoff after which to reject incoming blocks in this round (see {{<sref chain_sync>}}).
+- to maintain protocol liveness by allowing participants to try leader election in the next round if no one has produced a block in this round (see {{<sref storage_power_consensus>}}).
+
+In order to allow miners to do the above, the system clock must:
+1. have low clock drift: at most on the order of 2s (i.e. markedly lower than epoch time) at any given time.
```

Comment:
on the order of `1s` or `300ms`

Reply:
I think given a 15s block time, 2 would be fine. But will constrain to 1.